Automattic/wp-calypso
# Jetpack Error UX: Make side indicator use new Jetpack connection health check

Related to https://github.com/Automattic/dotcom-forge/issues/3459

## Proposed Changes

In this diff, I propose hooking the existing Jetpack error UI with data coming from a new API endpoint that checks Jetpack connection health as well as enable the indicator for Atomic sites.

![Screenshot 2023-08-15 at 14 03 49](https://github.com/Automattic/wp-calypso/assets/727413/daef9e5a-6044-42be-8cea-aad1de7468fb)

![Screenshot 2023-08-15 at 14 03 57](https://github.com/Automattic/wp-calypso/assets/727413/296169e0-ad29-4eec-8846-78cc3ffb45f5)

## Testing Instructions



1. Load the posts page for a blog with a broken connection
2. Confirm that the message in primary part of the page is displayed
3. Confirm that the error indicator is displayed in the sidebar's site card

## Pre-merge Checklist
